### PR TITLE
HDDS-6978. EC: Cleanup RECOVERING container on DN restarts

### PR DESCRIPTION
**Pull request by JacksonYao287**
## What changes were proposed in this pull request?

Cleanup RECOVERING container on DN restarts in containerReader

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-6978

### CODE REVIEW

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/ContainerReader.java [hunk 1]
```diff
@@ -207,6 +210,14 @@ public void verifyAndFixupContainerData(ContainerData containerData)
         KeyValueContainerUtil.parseKVContainerData(kvContainerData, config);
         KeyValueContainer kvContainer = new KeyValueContainer(kvContainerData,
             config);
+        if (kvContainer.getContainerState() == RECOVERING) {
+          if (shouldDeleteRecovering) {
+            kvContainer.delete();
```
Comment (umamaheswararao) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/ContainerReader.java:215]: Nit: Typo: delete --> Delete

Comment (umamaheswararao) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/ContainerReader.java:215]: Also I assume this delete will take care of cleaning the rocksDB entries as well.
CC: @guihecheng

File: hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/ozoneimpl/TestContainerReader.java [hunk 2]
```diff
@@ -217,7 +217,7 @@ private List<Long> addBlocks(KeyValueContainer keyValueContainer,
   @Test
   public void testContainerReader() throws Exception {
```
Comment (umamaheswararao) [inline hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/ozoneimpl/TestContainerReader.java:218]: How about we simply add a test which should simply create Recovering container and restart the DN and check?

Refer: TestContainerCommandsEC for creating recovering containers.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/ContainerReader.java (hunk 1 above)
Comment (JacksonYao287) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/ContainerReader.java:215]: yea, we should also delete the rocksDB entry of this container in merge rocksdb case

**Review by umamaheswararao — APPROVED** [review]
LGTM. @guihecheng do you have any comments before committing?
Pending CI too !

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/ContainerReader.java (hunk 1 above)
Comment (guihecheng) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/ContainerReader.java:215]: I think `kvContainer.delete()` already handles deleting kv pairs in the rocksdb for both before and after rocksdb-merge.

**Comment by guihecheng** [comment]
LGTM+1

**Comment by umamaheswararao** [comment]
Thanks a lot, @JacksonYao287 for the contribution!

**Pull request merged**